content/browser/tld_ephemeral_storage.cc

```diff
@@ -0,0 +1,90 @@
+/* Copyright (c) 2020 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/content/public/browser/tld_ephemeral_storage.h"
+
+#include <map>
+#include "base/no_destructor.h"
+#include "content/browser/dom_storage/dom_storage_context_wrapper.h"
+#include "content/browser/dom_storage/session_storage_namespace_impl.h"
+#include "content/browser/renderer_host/render_view_host_delegate.h"
+#include "content/browser/storage_partition_impl.h"
+#include "content/public/browser/render_view_host.h"
+#include "content/public/browser/session_storage_namespace.h"
+#include "content/public/browser/storage_partition.h"
+#include "content/public/browser/web_contents.h"
+#include "services/network/public/mojom/cookie_manager.mojom.h"
+
+namespace content {
+
+namespace {
+
+using TLDEphemeralStorageMap =
+    std::map<TLDEphemeralStorageKey, base::WeakPtr<TLDEphemeralStorage>>;
+
+// This map allows TLDEphemeralStorage to manage the lifetime of ephemeral
+// storage.
+TLDEphemeralStorageMap& active_tld_storage_areas() {
+  static base::NoDestructor<TLDEphemeralStorageMap> active_storage_areas;
+  return *active_storage_areas.get();
+}
+
+}  // namespace
+
+scoped_refptr<content::SessionStorageNamespace> CreateSessionStorageNamespace(
+    content::StoragePartition* partition,
+    const std::string& namespace_id) {
+  content::DOMStorageContextWrapper* context_wrapper =
+      static_cast<content::DOMStorageContextWrapper*>(
+          partition->GetDOMStorageContext());
+
+  return content::SessionStorageNamespaceImpl::Create(context_wrapper,
+                                                      namespace_id);
+}
+
+std::string GetSessionStorageNamespaceId(WebContents* web_contents) {
+  return web_contents->GetRenderViewHost()
+      ->GetDelegate()
+      ->GetSessionStorageNamespace(web_contents->GetSiteInstance())
+      ->id();
+}
+
+TLDEphemeralStorage::TLDEphemeralStorage(TLDEphemeralStorageKey key,
```

Comment:
why does this need to be in content?

Reply:
This class is the public interface that `EphemeralTabStorageHelper` uses to access private API in content. Is there a better place for this?

Reply:
what private api?

Reply:
the only private apis I see are accessed outside of this class and we're now adding two new patches for this. It would be better I think to just keep it all included in the browser_context override. Also brave/content should never have been created in the first place and the files that are already in there are misplaced

Reply:
if you created a separate file for tld_ephemeral_storage.cc/.h and included it from browser_context.cc you would put it in chromium_src

Reply:
There are three interfaces into private API here CreateSessionStorageNamespace and GetSessionStorageNamespaceId, which used to be in `BrowserContext` (still part of content). The final piece is `TLDEphemeralStorage` which talks to the `CookieStore` in order to manage ephemeral cookie lifetime. The idea is that eventually CreateSessionStorageNamespace and GetSessionStorageNamespaceId would become part of `TLDEphemeralStorage`.